Libraries/LibMarkdown/Table.cpp

```diff
@@ -0,0 +1,214 @@
+            original_length += span.text.length();
+        auto string = text.render_for_terminal(width);
+        format_builder.clear();
+        if (alignment == Alignment::Center) {
+            auto padding_length = (width - original_length) / 2;
+            builder.appendf("%*s%s%*s", padding_length, "", string.characters(), padding_length, "");
+            if ((width - original_length) % 2)
+                builder.append(' ');
+        } else {
+            format_builder.appendf("%%%s%zus", alignment == Alignment::Left ? "-" : "", width + (string.length() - original_length));
+            builder.appendf(
+                format_builder.to_string().characters(),
+                string.characters());
+        }
+    };
+
+    bool first = true;
+    for (auto& col : m_columns) {
+        if (!first)
+            builder.append('|');
+        first = false;
+        size_t width = col.relative_width * segment_width_proportion;
+        write_aligned(col.header, width, col.alignment);
+    }
+
+    builder.append("\n");
+    for (size_t i = 0; i < view_width; ++i)
+        builder.append('-');
+    builder.append("\n");
+
+    for (size_t i = 0; i < m_row_count; ++i) {
+        bool first = true;
+        for (auto& col : m_columns) {
+            if (col.rows.size() <= i)
+                continue;
+            auto& cell = col.rows[i];
+
+            if (!first)
+                builder.append('|');
+            first = false;
+
+            size_t width = col.relative_width * segment_width_proportion;
+            write_aligned(cell, width, col.alignment);
+        }
+        builder.append("\n");
+    }
+
+    return builder.to_string();
+}
+
+String Table::render_to_html() const
+{
+    StringBuilder builder;
+
+    builder.append("<table>");
+    builder.append("<thead>");
+    builder.append("<tr>");
+    for (auto& column : m_columns) {
+        builder.append("<th>");
+        builder.append(column.header.render_to_html());
+        builder.append("</th>");
+    }
+    builder.append("</tr>");
+    builder.append("</thead>");
+    builder.append("<tbody>");
+    for (size_t i = 0; i < m_row_count; ++i) {
+        builder.append("<tr>");
+        for (auto& column : m_columns) {
+            if (column.rows.size() > i) {
+                builder.append("<td>");
+                builder.append(column.rows[i].render_to_html());
+                builder.append("</td>");
+            }
+        }
+        builder.append("</tr>");
+    }
+    builder.append("</tbody>");
+    builder.append("</table>");
+
+    return builder.to_string();
+}
+
+OwnPtr<Table> Table::parse(Vector<StringView>::ConstIterator& lines)
+{
+    auto peek_it = lines;
+    auto first_line = *peek_it;
+    if (!first_line.starts_with('|'))
+        return nullptr;
+
+    ++peek_it;
+
+    if (peek_it.is_end())
+        return nullptr;
+
+    auto header_segments = first_line.split_view('|');
+    auto header_delimiters = peek_it->split_view('|');
```

Comment:
`keep_empty = true` on both, no?

Reply:
Hmmm...I somehow had this odd idea that `keep_empty` applied to the last element
yeah, sowwy~